data/routers/postDb-router.js

```diff
@@ -0,0 +1,135 @@
+const router = require('express').Router();
+
+const postDb = require('../helpers/postDb');
+const userDb = require('../helpers/userDb');
+
+router.post('/', (req, res) => {
+  const newPost = req.body;
+  if (newPost.text && newPost.user_id) {
+    userDb.getById(newPost.user_id).then(user => {
+      if (user) {
+        postDb
+          .insert(newPost)
+          .then(post => {
+            res.status(201).json(post);
+          })
+          .catch(err => {
+            res.status(500).json({
+              error: 'There was an error while saving the post to the database'
+            });
+          });
+      } else {
+        res
+          .status(404)
+          .json({ error: 'The user with the specified ID does not exist.' });
+      }
+    });
+  } else {
+    res.status(400).json({
+      error: 'Please provide a valid user_id and text for the post.'
+    });
+  }
+});
+
+router.get('/', (req, res) => {
+  postDb
+    .get()
+    .then(posts => {
+      res.status(200).json(posts);
+    })
+    .catch(err => {
+      res
+        .status(500)
+        .json({ error: 'The posts information could not be retrieved.' });
+    });
+});
+
+router.get('/:id', (req, res) => {
+  postDb
+    .getById(req.params.id)
+    .then(post => {
+      if (post) {
+        res.status(200).json(post);
+      } else {
+        res
+          .status(404)
+          .json({ error: 'The post with the specified ID does not exist.' });
+      }
+    })
+    .catch(err => {
+      res
+        .status(500)
+        .json({ error: 'The post information could not be retrieved.' });
+    });
+});
+
+// Since this function returns posts, I thought it belonged here in postDb-router.js rather than userDb-router.js
+router.get('/:id/all', (req, res) => {
+  userDb.getById(req.params.id).then(user => {
+    if (user) {
+      userDb
+        .getUserPosts(req.params.id)
+        .then(posts => {
+          res.status(200).json(posts);
+        })
+        .catch(err => {
+          res
+            .status(500)
+            .json({ error: 'The post information could not be retrieved.' });
+        });
+    } else {
+      res
+        .status(404)
+        .json({ error: 'The user with the specified ID does not exist.' });
+    }
+  });
+});
+
+router.delete('/:id', (req, res) => {
+  postDb
+    .remove(req.params.id)
+    .then(numberOfPosts => {
+      if (numberOfPosts === 1) {
+        res
+          .status(200)
+          .json({ message: `Number of records deleted: ${numberOfPosts}` });
+      } else {
+        res
+          .status(404)
+          .json({ error: 'The post with the specified ID does not exist.' });
+      }
+    })
+    .catch(err => {
+      res.status(500).json({ error: 'The post could not be removed' });
+    });
+});
+
+router.put('/', (req, res) => {
```

Comment:
shouldn't this be `/:id`? you pull the id off the the req.body just fine, but it's best practice for it to be sent along with the params. however, you should be verifying the presence of the `user_id` for the post instead